crates/arroyo-planner/src/tables.rs

```diff
@@ -511,16 +534,55 @@ impl Table {
                     Field::new(name, data_type, nullable),
                 );
 
-                let generating_expression = column.options.iter().find_map(|option| {
-                    if let ColumnOption::Generated {
-                        generation_expr, ..
-                    } = &option.option
-                    {
-                        generation_expr.clone()
-                    } else {
-                        None
-                    }
-                });
+                let generating_expression =
```

Comment:
This function has very deep nesting and the logic is quite hard to follow. Is there a way to simplify it?